### ISSUE TITLE
Really big package

### ISSUE THREAD

**Issue by Mark-H** [modmore/Commerce_mPDFWriter#1]
The package is over 120mb once uncompressed which is a little crazy. Most of that is fonts. 

Ideally, given the way packages are built with composer automatically being run before package, there would be a composer way of ignoring certain files/directories. But alas. 

Jako shared a similar solution he implemented in PDFResource.

https://github.com/Jako/PDFResource/blob/master/_build/config.json#L224-L229

https://github.com/theboxer/Git-Package-Management/blob/a495a4f9c82908dcbbbbb17b8c2746ec3f5ad302/core/components/gitpackagemanagement/processors/mgr/gitpackage/buildpackage.class.php#L55-L61

https://github.com/theboxer/Git-Package-Management/blob/a495a4f9c82908dcbbbbb17b8c2746ec3f5ad302/core/components/gitpackagemanagement/processors/mgr/gitpackage/buildpackage.class.php#L485-L506

Maybe that's something we can adapt.

**Comment by muzzwood** [comment]
Hmm interesting...
There's a bit of a description in the GPM docs too: http://theboxer.github.io/Git-Package-Management/config/build/#build-options-part

**Comment by muzzwood** [comment]
Update: appears to work fine! Package installed quickly and generated invoice looks good. I'll put together a PR for you later.

-----------

Here are some changes to the build script which result in the package only being 12MB.
I haven't had time to test it though - will get back to it ;)
```
<?php

/**
 * @param string $filename The name of the file.
 * @return string The file's content
 * @by splittingred
 */
function getSnippetContent($filename = '') {
    $o = file_get_contents($filename);
    $o = str_replace('<?php','',$o);
    $o = str_replace('?>','',$o);
    $o = trim($o);
    return $o;
}

function emptyFolder($path, $filemask = '*') {
    $inverse = false;
    if (strpos($filemask, '!') === 0) {
        $filemask = substr($filemask, 1);
        $inverse = true;
    }
    $files = glob($path . '/' . $filemask, GLOB_BRACE);
    if ($inverse) {
        $allFiles = glob($path . '/*');
        $files = array_diff($allFiles, $files);
    }
    foreach ($files as $file) {
        if (is_dir($file)) {
            emptyFolder($file, '*');
            rmdir($file);
        } else {
            unlink($file);
        }
    }
    return;
}

$mtime = microtime();
$mtime = explode(" ", $mtime);
$mtime = $mtime[1] + $mtime[0];
$tstart = $mtime;
set_time_limit(0);

if (!defined('MOREPROVIDER_BUILD')) {
    /* define version */
    define('PKG_NAME', 'Commerce_mPDFWriter');
    define('PKG_NAMESPACE', 'commerce_mpdfwriter');
    define('PKG_VERSION', '1.0.1');
    define('PKG_RELEASE', 'pl');

    /* load modx */
    require_once dirname(dirname(__FILE__)) . '/config.core.php';
    require_once MODX_CORE_PATH . 'model/modx/modx.class.php';
    $modx= new modX();
    $modx->initialize('mgr');
    $modx->setLogLevel(modX::LOG_LEVEL_INFO);
    $modx->setLogTarget('ECHO');


    echo '<pre>';
    flush();
    $targetDirectory = dirname(dirname(__FILE__)) . '/_packages/';
}
else {
    $targetDirectory = MOREPROVIDER_BUILD_TARGET;
}

$root = dirname(dirname(__FILE__)).'/';
$sources= array (
    'root' => $root,
    'build' => $root .'_build/',
    'events' => $root . '_build/events/',
    'resolvers' => $root . '_build/resolvers/',
    'validators' => $root . '_build/validators/',
    'data' => $root . '_build/data/',
    'plugins' => $root.'_build/elements/plugins/',
    'snippets' => $root.'_build/elements/snippets/',
    'source_core' => $root.'core/components/'.PKG_NAMESPACE,
    'source_assets' => $root.'assets/components/'.PKG_NAMESPACE,
    'lexicon' => $root . 'core/components/'.PKG_NAMESPACE.'/lexicon/',
    'docs' => $root.'core/components/'.PKG_NAMESPACE.'/docs/',
    'model' => $root.'core/components/'.PKG_NAMESPACE.'/model/',
);
unset($root);

$modx->loadClass('transport.modPackageBuilder','',false, true);
$builder = new modPackageBuilder($modx);
$builder->directory = $targetDirectory;
$builder->createPackage(PKG_NAMESPACE,PKG_VERSION,PKG_RELEASE);
$builder->registerNamespace(PKG_NAMESPACE,false,true,'{core_path}components/'.PKG_NAMESPACE.'/', '{assets_path}components/'.PKG_NAMESPACE.'/');

$modx->log(modX::LOG_LEVEL_INFO,'Packaged in namespace.'); flush();

$builder->package->put(
    [
        'source' => $sources['source_core'],
        'target' => "return MODX_CORE_PATH . 'components/';",
    ],
    [
        'vehicle_class' => 'xPDOFileVehicle',
        'validate' => [
            [
                'type' => 'php',
                'source' => $sources['validators'] . 'requirements.script.php'
            ]
        ],
        'resolve' => array(
            array(
                'type' => 'php',
                'source' => $sources['resolvers'] . 'loadmodules.resolver.php',
            )
        )
    ]
);

$modx->log(modX::LOG_LEVEL_INFO,'Packaged in resolvers.'); flush();

/* now pack in the license file, readme and setup options */
$builder->setPackageAttributes(array(
    'license' => file_get_contents($sources['docs'] . 'license.txt'),
    'readme' => file_get_contents($sources['docs'] . 'readme.txt'),
    'changelog' => file_get_contents($sources['docs'] . 'changelog.txt'),
));
$modx->log(modX::LOG_LEVEL_INFO,'Packaged in package attributes.'); flush();

// Specify directories to empty (mainly large fonts)
$emptyFolders = [
    $sources['source_core'].'/vendor/mpdf/mpdf/tmp' => "*",
    $sources['source_core'].'/vendor/mpdf/mpdf/ttfonts' => "!{DejaVu*,Free*}",
    $sources['source_core'].'/vendor/mpdf/mpdf/ttfontdata' => "*"
];
if (!empty($emptyFolders)) {
    foreach ($emptyFolders as $emptyFolder => $emptyFiles) {
        emptyFolder($emptyFolder, $emptyFiles);
    }
    $modx->log(modX::LOG_LEVEL_INFO,'Removed unused fonts from mPDF.'); flush();
}

$modx->log(modX::LOG_LEVEL_INFO,'Packing...'); flush();
$builder->pack();

$mtime = microtime();
$mtime = explode(" ", $mtime);
$mtime = $mtime[1] + $mtime[0];
$tend = $mtime;
$totalTime = ($tend - $tstart);
$totalTime = sprintf("%2.4f s", $totalTime);

$modx->log(modX::LOG_LEVEL_INFO,"\n<br />Package Built.<br />\nExecution time: {$totalTime}\n");


```

**Comment by muzzwood** [comment]
See PR  #2

**Comment by Mark-H** [comment]
Fixed in v1.1 (#2)
